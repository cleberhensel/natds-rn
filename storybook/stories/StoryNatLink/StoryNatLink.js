import React from 'react';
import {
  ScrollView,
  SafeAreaView,
  Dimensions,
} from 'react-native';
import {
  NatContainer,
  NatSpace,
  ThemeNaturaLight as Theme,
  NatText,
  NatLink,
} from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatLink = () => (
  <SafeAreaView style={styles.defaultScreen}>
    <NatContainer>
      <ScrollView>
        <NatSpace
          marginTop="small"
          marginBottom="none"
          marginRight="none"
          marginLeft="none"
          paddingTop="none"
          paddingBottom="standard"
          paddingRight="none"
          paddingLeft="none"
        />
        <NatText value="Text without link" type="body1" />
        <NatLink value="Text with link" type="body1" onPress={() => alert('click link')} />
        <NatLink value="Text with link" type="body1" onPress={() => alert('click link')} align="center" />
        <NatLink value="Text with link" type="body1" onPress={() => alert('click link')} align="flex-end" />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>

);
