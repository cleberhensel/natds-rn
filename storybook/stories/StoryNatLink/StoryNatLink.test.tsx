import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components/native";
import { StoryNatLink } from "./StoryNatLink";
describe("StoryNatLink component", () => {
  it("Should render StoryNatLink component ", () => {
    const button = renderer.create(<StoryNatLink />).toJSON();
    expect(button).toMatchSnapshot("StoryNatLink snapshot");
  });
});
