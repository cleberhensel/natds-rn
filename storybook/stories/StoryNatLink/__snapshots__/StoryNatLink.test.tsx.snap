// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatLink component Should render StoryNatLink component : StoryNatLink snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }
>
  <View
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="small"
    paddingLeft="small"
    paddingRight="small"
    paddingTop="small"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          backgroundColor="transparent"
          marginBottom="none"
          marginLeft="none"
          marginRight="none"
          marginTop="small"
          paddingBottom="standard"
          paddingLeft="none"
          paddingRight="none"
          paddingTop="none"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 16,
                "paddingBottom": 24,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        />
        <View
          margin={24}
          style={
            Array [
              Object {
                "marginBottom": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <Text
            color="#333333"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
              ]
            }
            type="body1"
          >
            Text without link
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "opacity": 1,
            }
          }
        >
          <View
            margin={24}
            style={
              Array [
                Object {
                  "marginBottom": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#227BBD"
              style={
                Array [
                  Object {
                    "color": "#227BBD",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                ]
              }
              type="body1"
            >
              Text with link
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "opacity": 1,
            }
          }
        >
          <View
            margin={24}
            style={
              Array [
                Object {
                  "marginBottom": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#227BBD"
              style={
                Array [
                  Object {
                    "color": "#227BBD",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                ]
              }
              type="body1"
            >
              Text with link
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-end",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "opacity": 1,
            }
          }
        >
          <View
            margin={24}
            style={
              Array [
                Object {
                  "marginBottom": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <Text
              color="#227BBD"
              style={
                Array [
                  Object {
                    "color": "#227BBD",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                ]
              }
              type="body1"
            >
              Text with link
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
